python-package/lightgbm/basic.py

```diff
@@ -1115,13 +1188,144 @@ def __init__(self, data, label=None, reference=None,
         self.feature_penalty = None
         self.monotone_constraints = None
         self.version = 0
+        self._start_row = 0  # Used when pushing rows one by one.
 
     def __del__(self):
         try:
             self._free_handle()
         except AttributeError:
             pass
 
+    def create_sample_indices(self, total_nrow):
+        """Create sample indices for the given parameter of the Dataset.
```

Comment:
```suggestion
        """Get an array of randomly chosen indices from this ``Dataset``.
        
        Indices are sampled without replacement.
```

I was confused by this description. Would you consider this change? Hopefully I've understood this method correctly.